test/integration/unlogged_in_registered_user_must_login_to_checkout_test.rb

```diff
@@ -0,0 +1,35 @@
+require 'test_helper'
+
+class UnloggedInRegisteredUserMustLoginToCheckoutTest < ActionDispatch::IntegrationTest
+
+  test 'that an a registered user must log in to checkout' do
+    create_category_and_items(1)
+    create_user
+    visit items_path
+    item1 = Item.find_by(title: "pour over0")
+
+    within "#item#{item1.id}" do
+      click_button "Add to Cart"
+    end
+
+    visit "/cart"
+    click_button "Checkout"
+    assert "/login", current_path
+
+    assert page.has_content?("Must Login")
+     fill_in 'Username', with: 'ryan'
```

Comment:
Inconsistent indentation detected.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.